sklearn/metrics/pairwise.py

```diff
@@ -256,6 +277,150 @@ def euclidean_distances(X, Y=None, Y_norm_squared=None, squared=False,
     return distances if squared else np.sqrt(distances, out=distances)
 
 
+# Pairwise distances in the presence of missing values
+def masked_euclidean_distances(X, Y=None, squared=False,
+                               missing_values="NaN", copy=True):
+    """Calculates euclidean distances in the presence of missing values
+
+    Considering the rows of X (and Y=X) as vectors, compute the
+    distance matrix between each pair of vectors. Similarly, if
+    Y is not X, then compute the distance matrix between each
+    pair of vectors (i.e., each row pair) in X and Y.
+
+    This function computes pairwise euclidean distance for vectors
+    in dense matrices X and Y with missing values in arbitrary
+    coordinates.
+
+    The following formula is used for this:
```

Comment:
Sorry, is not noticed. I have tried to suggest that the matrix formulation here is unhelpful. You just need enough for the intuition behind calculating the metric to be clear. A couple of sentences